puppet/modules/jenkins_job_builder/files/centos.org/jobs/foreman-test.yml

```diff
@@ -0,0 +1,75 @@
+---
+- job-template:
+    name: foreman-{version}-{distro}-{action}-test
+    project-type: 'pipeline'
+    sandbox: true
+    parameters:
+      - string:
+          name: expected_version
+          default: 'UNKNOWN'
+      - string:
+          name: version
+          default: '{version}'
+      - string:
+          name: distro
+          default: '{distro}'
+      - string:
+          name: action
+          default: '{action}'
+      - string:
+          name: type
+          default: 'foreman'
+    dsl:
+      !include-raw:
+        - pipelines/lib/foremanCentosJob.groovy{empty}
+        - pipelines/lib/pipelines.groovy{empty}
+        - ../theforeman.org/pipelines/lib/duffy.groovy{empty}
+        - ../theforeman.org/pipelines/lib/ansible.groovy{empty}
+        - ../theforeman.org/pipelines/lib/foreman_infra.groovy{empty}
+
+- project:
+    name: foreman-release-test
+    jobs:
+      - 'foreman-{version}-{distro}-{action}-test'
+    empty: ''
+    distro:
+      - centos7
+      - debian9
+      - debian10
+      - ubuntu1604
+      - ubuntu1804
+    version:
+      - '1.23'
+      - '1.24'
+      - '2.0'
```

Comment:
Some combos of this are invalid. See https://github.com/theforeman/forklift/blob/master/vagrant/config/versions.yaml as well. Like Debian 9 / Ubuntu 16.04 + Foreman 2.0.